packages/settingregistry/src/plugin-schema.json

```diff
@@ -15,26 +33,103 @@
       "type": "string",
       "default": ""
     },
-    "jupyter.lab.setting-icon-label": { "type": "string", "default": "Plugin" },
+    "jupyter.lab.setting-icon-label": {
+      "type": "string",
+      "default": "Plugin"
+    },
     "jupyter.lab.shortcuts": {
-      "items": { "$ref": "#/definitions/shortcut" },
+      "items": {
+        "$ref": "#/definitions/shortcut"
+      },
       "type": "array",
       "default": []
     },
-    "jupyter.lab.transform": { "type": "boolean", "default": false }
+    "jupyter.lab.transform": {
+      "type": "boolean",
+      "default": false
+    }
   },
   "definitions": {
+    "menu": {
+      "properties": {
+        "disabled": {
+          "type": "boolean",
+          "default": false
+        },
+        "id": {
+          "type": "string"
```

Comment:
akin to the typescript side of the house, but here you can help people avoid the foot gun of unselectable ids (since i trust these become real DOM ids?):

```json
"oneOf": [
   {"type": "string", "enum": ["jp-menu-file", "..."]},
   {"type": "string", "pattern": "[a-z][a-z0-9\\-_]+"}
]
```